subxt/src/config/mod.rs

```diff
@@ -51,6 +51,9 @@ pub trait Config: Sized + Send + Sync + 'static {
 
     /// This type defines the extrinsic extra and additional parameters.
     type ExtrinsicParams: ExtrinsicParams<Self>;
+
+    /// Asset Id
+    type AssetId: Debug + Encode;
```

Comment:
This _could_ also impl `EncodeAsType` possibly (same as a couple of the others above like singature I think), which would at least mean that if somebody tried setting a tip with an asset ID, and it was configured wrong, we'd get an `EncodeAsType` error back (eg because it failed to encode an `Option<u32>` into an `Option<MultiLocation>`) rather than have some error from the node that it failed to decode the tx provided.

For now I'd leave it as is, but we should probably consider moving some things in Config to `EncodeAsType` instead of `Encode` so that we get this extra protection.

Reply:
Might be worth an issue to not forget about it 🙏